xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -1873,8 +1873,12 @@ void COutput::Flush()
   {
     if (msg->signal == COutputDataProtocol::NEWFRAME)
     {
-      CVaapiDecodedPicture pic = *reinterpret_cast<CVaapiDecodedPicture*>(msg->data);
-      m_config.videoSurfaces->ClearRender(pic.videoSurface);
+      CPayloadWrap<CVaapiDecodedPicture> *payload;
+      payload = dynamic_cast<CPayloadWrap<CVaapiDecodedPicture>*>(msg->payloadObj.get());
```

Comment:
declaration and assignment can be done in the same line

Reply:
I know, I'd always do it that way (and use `auto`), but this is copy'n'pasted from 61dfb26640761e217cd192529fa65fae2117aa2c so I figured this was the common Kodi coding style. Do you want me to change it?

Reply:
Just keep it like you have done it now.
There is no strict rule for deceleration and assignment, but in usually it is done in one line.